core/src/main/java/org/opensearch/sql/data/model/ExprDateValue.java

```diff
@@ -25,20 +32,44 @@
 public class ExprDateValue extends AbstractExprValue {
 
   private final LocalDate date;
+  private String datePattern;
+  // List of cleaned patterns
+  private static final List<String> CLEANED_PATTERNS = Arrays.stream(DATE_TIME_FORMATTER_PATTERNS_NANOS_OPTIONAL.split("\\]\\["))
+          .map(pattern -> pattern.replace("[", "").replace("]", ""))
+          .collect(Collectors.toList());
 
   /** Constructor of ExprDateValue. */
   public ExprDateValue(String date) {
     try {
       this.date = LocalDate.parse(date, DATE_TIME_FORMATTER_VARIABLE_NANOS_OPTIONAL);
+      this.datePattern = determineDatePattern(date);
```

Comment:
does parese execute 2 times, the reasons is LocalDate.parse already parse the date string

Reply:
On line #44 here, yes we are already parsing the date string to Local date object. But I need to know the exact pattern that matched the input date string. So inside `determineDatePattern` function, using `parse` function (LocalDate/LocalDatetime) I'm determining the matching pattern and returning that to set `this.datePattern`. This date pattern is used to format the LocalDate date object instead of default ISO_LOCAL_DATE string inside `value()`.